src/libpmemfile/preload.c

```diff
@@ -1636,3 +1780,153 @@ hook_fchownat(struct fd_desc at, const char *path,
 
 	return check_errno(r);
 }
+
+static long
+hook_sendfile(long out_fd, long in_fd, off_t *offset, size_t count)
+{
+	if (fd_pool_has_allocated(out_fd))
+		return check_errno(-ENOTSUP);
+
+	if (fd_pool_has_allocated(in_fd))
+		return check_errno(-ENOTSUP);
+
+	return syscall_no_intercept(SYS_sendfile, out_fd, in_fd, offset, count);
+}
+
+static long
+hook_readlinkat(struct fd_desc at, const char *path,
+				char *buf, size_t bufsiz)
+{
+	struct resolved_path where;
+
+	resolve_path(at, path, &where, NO_RESOLVE_LAST_SLINK);
+
+	if (where.error_code != 0)
+		return where.error_code;
+
+	if (where.at.pmem_fda.pool == NULL)
+		return syscall_no_intercept(SYS_readlinkat,
+		    where.at.kernel_fd, where.path, buf, bufsiz);
+
+	ssize_t r = pmemfile_readlinkat(where.at.pmem_fda.pool->pool,
+			where.at.pmem_fda.file, where.path, buf, bufsiz);
+
+	if (r != 0)
+		r = -errno;
+
+	log_write("pmemfile_readlinkat(%p, %p, \"%s\", \"%s\", %zu) = %zd",
+	    (void *)where.at.pmem_fda.pool->pool,
+	    (void *)where.at.pmem_fda.file,
+	    where.path, buf, bufsiz, r);
+
+	return check_errno(r);
+}
+
+static long
+nosup_syscall_with_path(long syscall_number,
+			long path, int resolve_last,
+			long arg0, long arg1,
+			long arg2, long arg3,
+			long arg4, long arg5)
+{
+	struct resolved_path where;
+
+	resolve_path(cwd_desc(), (const char *)path, &where, resolve_last);
+
+	if (where.error_code != 0)
+		return where.error_code;
+
+	/* XXX */
```

Comment:
It is my problem with path resolution, and _at interfaces ( or lack of them sometimes ).
I'll describe it in a comment, maybe tomorrow -- I need to find good way to explain it.